src/components/subscriptions/expiration/SubscriptionExpirationBanner.jsx

```diff
@@ -70,28 +71,44 @@ const SubscriptionExpirationBanner = ({ isSubscriptionPlanDetails }) => {
     return null;
   }
 
+  let subscriptionExpirationThreshold = SUBSCRIPTION_DAYS_REMAINING_MODERATE;
   let dismissible = true;
   let alertType = 'info';
   if (daysUntilExpiration <= SUBSCRIPTION_DAYS_REMAINING_SEVERE) {
+    subscriptionExpirationThreshold = SUBSCRIPTION_DAYS_REMAINING_SEVERE;
     alertType = 'warning';
   }
   if (daysUntilExpiration <= SUBSCRIPTION_DAYS_REMAINING_EXCEPTIONAL) {
+    subscriptionExpirationThreshold = SUBSCRIPTION_DAYS_REMAINING_EXCEPTIONAL;
     dismissible = false;
     alertType = 'danger';
   }
 
+  const emitBannerActionEvent = (dismiss) => {
```

Comment:
[nit/suggestion] It might be worth splitting this function into two: `emitAlertActionEvent` and `emitAlertDismissedEvent` so it's a bit more readable at a glance without having to decipher the `true/false` flag.